src/general/StringUtils.cs

```diff
@@ -41,4 +42,49 @@ public static string FormatNumber(this double number, bool withSuffix = true)
 
         return number.ToString("0.#", CultureInfo.CurrentCulture);
     }
+
+    /// <summary>
+    ///   Splits string into different chunks by whitespace.
+    /// </summary>
+    /// <param name="input">String to split.</param>
+    /// <param name="ignoreWithinQuotes">Ignore whitespaces within quotation marks.</param>
+    public static string[] SplitByWhiteSpace(string input, bool ignoreWithinQuotes)
```

Comment:
I *think* we may need a way to escape `"`'s within a string so you could have text like `[thrive:compound text="This is some \"stuff\" in the game"]` And of course double `\` would mean then a single `\`.

Reply:
Added.